packages/status-js/src/client/client.ts

```diff
@@ -111,18 +136,36 @@ class Client {
     await this.waku.stop()
   }
 
-  public createAccount = async (): Promise<Account> => {
-    this.account = new Account()
+  get account() {
+    return this.#account
+  }
+
+  set account(account: Account | undefined) {
+    this.#account = account
+    this.storage.setItem(THROWAWAY_ACCOUNT_STORAGE_KEY, this.#account)
+
+    for (const listener of this.#accountListeners) {
+      listener(this.#account ?? undefined)
+    }
+  }
 
-    await this.community.requestToJoin()
+  public createAccount(): Account {
+    this.account = new Account(this)
+    this.account.updateMembership(this.community)
```

Comment:
non-blocking: if you agree with me on https://github.com/status-im/status-web/pull/315#discussion_r988240407, I wouldn't handle `requested` state in `handleDescription`, but do it

1. `community.requestToJoin()`
2. `this.membership = 'requested'`
3. `this.account = this`

here instead.

Since, personally, I don't look at it as a "push event" like `rejected` and `approved`. And I also see that `deleteAccount` sets the account within its block (scope).

Reply:
It was implemented this way initially. However, to know whether you should even perform a request to join, you need to know whether the account did not become a member in the meantime and perform the logic already existing `updateMembership` method. I didn't want this to leak outside of the `Account` itself.

Reply:
> Also, I have a proposal for a re-architecture to allow this decoupling better.
> – https://github.com/status-im/status-web/pull/315#discussion_r989002168